packages/base/src/state/bond_providers.rs

```diff
@@ -0,0 +1,232 @@
+use crate::error::core::{ContractError, ContractResult};
+use cosmwasm_std::{Addr, Empty, StdResult, Storage};
+use cw_storage_plus::{Item, Map};
+
+pub struct BondProviders<'a> {
+    pub providers: Map<'a, Addr, Empty>,
+    pub next_provider_ptr: Item<'a, u64>,
+}
+
+impl<'a> BondProviders<'a> {
+    pub const fn new(storage_key: &'a str) -> Self {
+        Self {
+            providers: Map::new(storage_key),
+            next_provider_ptr: Item::new("bond_providers_ptr"),
+        }
+    }
```

Comment:
This API is dangerous. If user creates more than a single BondProviders instance, then next_provider_ptr will be dangling between them.